File: printerservice/src/main/java/com/example/print/printerserver/java_connectors/FilePrinterConnector.java
package com.example.print.printerserver.java_connectors;

import java.io.DataOutputStream;
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;

public class FilePrinterConnector extends PrinterConnector{

    private File file;

    public FilePrinterConnector(File file) {
        this.file = file;
    }

    @Override
    public void writeData(DataOutputStream out) throws IOException {
        FileInputStream inputStream = new FileInputStream(file);
        byte[] buffer = new byte[3000];
        while (inputStream.read(buffer) != -1)
            out.write(buffer);
    }
}

File: printerservice/src/main/java/com/example/print/printerserver/connectors/FilePrinterConnector.kt
package com.example.print.printerserver.connectors

import com.example.print.printerserver.utils.write
import java.io.DataOutputStream
import java.io.File


internal class FilePrinterConnector(private val data: File) : PrinterConnector() {

    override fun writeData(output: DataOutputStream?) {
        output?.write(data)
    }
}